tlatools/org.lamport.tlatools/src/tlc2/value/ValueInputStream.java

```diff
@@ -91,7 +91,27 @@ public final IValue read() throws IOException {
 		}
 	}
 	
+	/**

```

Comment:
Perhaps, this (and https://github.com/tlaplus/tlaplus/pull/1088#discussion_r1876749598) suggests that the `IOUtils` module should be migrated to TLC,  like the `Json` module was eventually [migrated](https://github.com/tlaplus/CommunityModules/pull/84).

Reply:
I see, interesting. I agree. This feature is strongly coupled, and having it in 2 codebases doesn't make sense.

I'm not sure what this migration process would be, however.

It seems maybe:
1. bundle `IOUtils.tla` at some point in time (now, later?)
2. at some future point, make that feature built-in?

Reply:
The Json module and its Java module override was just copied over to TLC (modifications have been applied to both repositories).

Reply:
Oh ok, and then the assumption is that people will download a new TLC and a new CommunityModules at the same time... (right?)

I could do that in this PR then. That would make testing nicer, certainly (see other conversation).

Reply:
[Yes, the other alternative is to make the CM a test-time dependency of TLC. https://github.com/tlaplus/CommunityModules/commit/e43de9c902a1d84744509592d9380c4bd9335751 makes sure that `IOUtils` can be used from ](https://github.com/tlaplus/tlaplus/pull/1103)